wgpu/src/image/raster.rs

```diff
@@ -132,3 +145,67 @@ impl Cache {
         self.map.contains_key(&handle.id())
     }
 }
+
+#[derive(Debug, Clone, Copy)]
+enum Orientation {
+    Default,
+    FlippedHorizontally,
+    FlippedVertically,
+    Rotated90,
+    Rotated180,
+    Rotated270,
+    Rotated90AndFlippedHorizontally,
+    Rotated90AndFlippedVertically,
+}
+
+impl Orientation {
+    // Meaning of the returned value is described e.g. at:
+    // https://magnushoff.com/articles/jpeg-orientation/
+    fn from_exif<R>(reader: &mut R) -> Result<Self, exif::Error>
+    where
+        R: std::io::BufRead + std::io::Seek,
+    {
+        let exif = exif::Reader::new().read_from_container(reader)?;
+
+        Ok(exif
+            .get_field(::exif::Tag::Orientation, ::exif::In::PRIMARY)
```

Comment:
Should we drop the `::` prefix also here and on l.188? I thought this one here would get confused with `exif` var defined on l.168, but I just tried and apparently it works (I'm still absolutely a Rust newbie)